File: uva10038/uva10038.py
import sys


def isJolly(lista):


	tamanho = int (lista[0])

	listaDiferencaAbsoluta = []

	i = 1
	while (i < tamanho):
		# print(lista[i], lista[i+1])
		listaDiferencaAbsoluta.append(abs(lista[i] - lista[i+1]))
		i += 1

	# print(listaDiferencaAbsoluta)

	listaDiferencaAbsoluta.sort(reverse = True)
	# print(listaDiferencaAbsoluta,end="")

	if (len(listaDiferencaAbsoluta) == 0 ):
		return True
	else:


		if (listaDiferencaAbsoluta[-1] != 1 or listaDiferencaAbsoluta[0] != (tamanho - 1)):
			return False
			
		for i in range (len(listaDiferencaAbsoluta)-1):

			if (abs(listaDiferencaAbsoluta[i] - listaDiferencaAbsoluta[i+1]) > 1) :

				return False

		return True


def main():

	linha = sys.stdin.readline()
	while ( linha != ""):

		lista = []
		for elemento in linha.split(): 
			lista.append( int(elemento) )

		resposta = "Jolly" if isJolly(lista) else "Not jolly"
		print(resposta)

		linha = sys.stdin.readline()

	return 0

main() 

File: uva10038/uva10038.java
/* Main.java
 * UVA 10038 - Jolly Jumpers
 * Autores: Jackson Willian
 */

import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;
import java.lang.Math;

class Main {

	public static void main(String[] args) {
		
		Scanner scanner;  
		String linha, resposta;
		String[] vetorNumeros;

		scanner = new Scanner(System.in);
		while( scanner.hasNext() ) {
			linha = scanner.nextLine();
			vetorNumeros  = linha.split(" ");
			resposta = isJolly(vetorNumeros) ? "Jolly" : "Not jolly";
		    System.out.println( resposta );
		}
		scanner.close();
	}


	public static boolean isJolly (String[] vetorNumeros){
		
		ArrayList<Integer> listaDiferenca = new ArrayList<Integer>();
		int tamanhoVetor, tamanhoLista, i, valorA, valorB;

		tamanhoVetor = Integer.parseInt(vetorNumeros[0]);
		i = 1;
		while (i < tamanhoVetor){
			valorA = Integer.parseInt(vetorNumeros[i]);
			valorB = Integer.parseInt(vetorNumeros[i+1]);
			listaDiferenca.add( Math.abs(valorA - valorB) );
			i++;
		}

		Collections.sort( listaDiferenca, Collections.reverseOrder() );

		tamanhoLista = listaDiferenca.size();
		if ( tamanhoLista == 0){
			return true;
		}else if ( (listaDiferenca.get( tamanhoLista - 1 ) != 1) || (listaDiferenca.get( 0 ) != (tamanhoVetor - 1) )) {
			return false;
		}else{
			for(i = 0; i < (tamanhoLista - 1); i++ ){
				if( Math.abs( listaDiferenca.get(i) - listaDiferenca.get(i+1) ) > 1 ){
					return false;
				}
			}
			return true;
		}
	}
}

